src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1506 @@
+        diagrams = cat.ID
+        diagrams_cat = cat
+      elseif string.lower(cat:CreateName().LuaString) == "harp pedals" and diag == false then
+        print("Found Harp Pedals category")
+        diagrams = cat.ID
+        diagrams_cat = cat
+      end -- if cat...
+    end -- for cat
+-----
+    -- find an existing diagram 
+    --
+    textexpressiondefs:LoadAll()
+    for ted in each(textexpressiondefs) do
+      if ted.CategoryID == diagrams and ted:CreateDescription().LuaString == description.LuaString then
+        print ("Diagram found at",ted.ItemNo)
+        diag_ted = ted.ItemNo
+      end -- if ted.CategoryID
+    end -- for ted...
+    --
+    -- if there is no existing diagram...', create one
+    if diag_ted == 0 then
+      local ex_ted = finale.FCTextExpressionDef()
+      local ted_text = finale.FCString()
+      if diag then
+        local text_font = diagram_font
+        ted_text.LuaString = text_font..diagram_string.LuaString
+      else 
+        ted_text.LuaString = diagram_string.LuaString
+      end -- if diag
+      ex_ted:AssignToCategory(diagrams_cat)
+      ex_ted:SetDescription(description)
+      ex_ted:SaveNewTextBlock(ted_text)
+      if use_tech then -- If using the Techniques category, override positioning
+        ex_ted:SetUseCategoryPos(false)
+        --ex_ted.HorizontalJustification = 1 -- Justify Center
+        --ex_ted.HorizontalAlignmentPoint = 5 -- Center on Music
+        --ex_ted.HorizontalOffset = nudge
+        ex_ted.VerticalAlignmentPoint = 9 -- align to 
+        ex_ted.VerticalBaselineOffset = 12
+        ex_ted.VerticalEntryOffset = -36
+      end
+      ex_ted:SaveNew()
+      diag_ted = ex_ted.ItemNo
+    end -- if diag_ted == 0
+
+-- Test to see if diagram is there already...
+    expressions:LoadAllForRegion(region)
+    for e in each(expressions) do
+      local ted = e:CreateTextExpressionDef()
+      local ted_desc = ted:CreateDescription()
+      if ted_desc:ContainsLuaString(desc_prefix.LuaString) then
+        diag_asn = true
+        goto error1
+      end 
+      end -- for e in expressions... ]]
+
+-- add the harp diagram
+
+      add_expression:SetStaff(staff_num)
+      add_expression:SetMeasurePos(measure_pos)
+      add_expression:SetID(diag_ted)
+      add_expression:SaveNewToCell(and_cell)
+
+-------
+      ::error1::
+      finenv.EndUndoBlock(true)
+      if error == true then -- Tried this as if error then, it broke the script
+        print("There seems to be a problem with your harp diagram.")
+        local result = ui:AlertYesNo("There seems to be a problem with your harp diagram. \n Would you like to try again?", NULL)
+--                if result == 2 then harp_dialog() end
+      end -- error
+      if diag_asn then
+        ui:AlertInfo("There is already a harp diagram assigned to this region.", NULL)
+      end
+    end -- function add_pedals
+
+    -------
+    function harp_scale(root, scale, diag, chd, partial)
```

Comment:
It would be best to have more descriptive variable names—especially for function arguments. I assume `diag` means diagram? However, I have no idea what `chd` is from looking at the code.